### DISCUSSION TITLE
Run test keyboard shortcut ( linux )

### DISCUSSION THREAD

**Discussion by maciejbak85** [scalameta/metals#2992] [Q&A]
Hi,

I am trying to find keyboard shortcut for running unit test.
When I try ctrl+A I see a message: "No debuggable tests found in this workspace. You may need to install a test provider extension" , probably some wrong shortcut.

**Comment by tgodzik** [comment]
Thanks for the question! This looks like maybe the Metals extension wasn't activated? The message is coming from VS Code itself.

**Reply by maciejbak85** [reply]
it is activated, I am using it for a longer time ;)

**Reply by tgodzik** [reply]
Hmm... so maybe it's a wrong shortcut then. There is a couple of ways to run the tests

- using [launch.json](https://scalameta.org/metals/docs/editors/vscode#via-a-launchjson-configuration)  and then you can either choose with `workbench.action.debug.selectandstart` command or via F5
- using Metals commands such as `metals.test-current-file` or `metals.test-current-target`- these can be bound to shortcuts
- code lenses

**Reply by maciejbak85** [reply]
forgot to reply:

```
{
  "key": "f1",
  "command": "metals.test-current-file"
}
```

helped :) thanks
